manifests-gen/providers.go

```diff
@@ -102,17 +104,52 @@ func (p *provider) writeProviderComponentsToManifest(fileName string, objs []uns
 	return os.WriteFile(path.Join(*manifestsPath, fileName), ensureNewLine(combined), 0600)
 }
 
-// writeProviderComponentsConfigmap allows to write provider components to the provider (transport) ConfigMap.
-func (p *provider) writeProviderComponentsConfigmap(fileName string, objs []unstructured.Unstructured) error {
-	combined, err := utilyaml.FromUnstructured(objs)
+// compressToZstd compressses bytes using zstd
+func (p *provider) compressToZstd(data []byte) (bytes.Buffer, error) {
+	var compressed bytes.Buffer
+
+	writer, err := zstd.NewWriter(&compressed, zstd.WithEncoderLevel(zstd.EncoderLevelFromZstd(19)))
```

Comment:
Could we use the [exposed constants](https://github.com/klauspost/compress/blob/7ae2138b16cc43afcea3ce7d3d2f2625fb389d51/zstd/encoder_options.go#L149-L170) instead of the number here?

Also I'd use the fastest option, to improve decompression time, as we will do it quite often during reconciles

Reply:
Changed to SpeedBestCompression. In zstd the compression level does not impact decompression speed.

Reply:
Gotcha, thanks for clarifying!